packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../convert.dart';
+import 'logger.dart';
+
+class SizeAnalyzer {
+  SizeAnalyzer(this._fileSystem, this._logger, this._processUtils);
+
+  final FileSystem _fileSystem;
+  final Logger _logger;
+  final ProcessUtils _processUtils;
+
+  static const String aotSizeFileName = 'aot-size.json';
+
+  static const int tableWidth = 80;
+
+  Future<Map<String, dynamic>> analyzeApkSize({
```

Comment:
add dart doc describing what this method does

Reply:
Done